### ISSUE TITLE
bazel: unable to test pkg/ccl/importccl due to missing test data read permissions

### ISSUE THREAD

**Issue by irfansharif** [cockroachdb/cockroach#56093]
Trying to test pkg/importccl. Needs to pick up https://github.com/cockroachdb/cockroach/pull/56092 first.

```
➜  cockroach git:(master) bazel test //pkg/ccl/importccl:all
INFO: Analyzed 2 targets (76 packages loaded, 935 targets configured).
INFO: Found 1 target and 1 test target...
FAIL: //pkg/ccl/importccl:importccl_test (see /private/var/tmp/_bazel_irfansharif/99e666e4e674209ecdb66b46371278df/execroot/cockroach/bazel-out/darwin-fastbuild/testlogs/pkg/ccl/importccl/importccl_test/test.log)
INFO: Elapsed time: 91.558s, Critical Path: 83.58s
INFO: 44 processes: 3 internal, 41 darwin-sandbox.
INFO: Build completed, 1 test FAILED, 44 total actions
//pkg/ccl/importccl:importccl_test                                       FAILED in 65.9s
  /private/var/tmp/_bazel_irfansharif/99e666e4e674209ecdb66b46371278df/execroot/cockroach/bazel-out/darwin-fastbuild/testlogs/pkg/ccl/importccl/importccl_test/test.log
INFO: Build completed, 1 test FAILED, 44 total actions
```

```
➜  cockroach git:(master) cat  /private/var/tmp/_bazel_irfansharif/99e666e4e674209ecdb66b46371278df/execroot/cockroach/bazel-out/darwin-fastbuild/testlogs/pkg/ccl/importccl/importccl_test/test.log
exec ${PAGER:-/usr/bin/less} "$0" || exit 1
Executing tests from //pkg/ccl/importccl:importccl_test
-----------------------------------------------------------------------------
I201028 18:32:54.425274 1 (gostd) rand.go:85  random seed: 2084959252895124278
n1: found liveness
n1: found liveness
test logs left over in: /var/folders/3j/6fxg9ckn21g5xm0z_06r7gzr0000gn/T/logTestImportCSVStmt339012350
--- FAIL: TestImportCSVStmt (1.18s)
    test_log_scope.go:154: test logs captured to: /var/folders/3j/6fxg9ckn21g5xm0z_06r7gzr0000gn/T/logTestImportCSVStmt339012350
    test_log_scope.go:63: use -show-logs to present logs inline
    import_stmt_test.go:1431: open testdata/csv/table: operation not permitted
test logs left over in: /var/folders/3j/6fxg9ckn21g5xm0z_06r7gzr0000gn/T/logTestImportIntoCSV603611106
--- FAIL: TestImportIntoCSV (1.03s)
    test_log_scope.go:154: test logs captured to: /var/folders/3j/6fxg9ckn21g5xm0z_06r7gzr0000gn/T/logTestImportIntoCSV603611106
    test_log_scope.go:63: use -show-logs to present logs inline
    import_stmt_test.go:2309: open testdata/csv/empty.csv: operation not permitted
n1: found liveness
FAIL
```

Looks like we don't have read permissions for `testdata/csv/empty.csv` within the sandbox.

**Comment by miretskiy** [comment]
I suspect it's an issue due to non-hermetic tests:

```
// Table schema used in IMPORT TABLE tests.
	tablePath := filepath.Join(baseDir, "table")
	if err := ioutil.WriteFile(tablePath, []byte(`
		CREATE TABLE t (
			a int8 primary key,
			b string,
			index (b),
			index (a, b)
		)
	`), 0666); err != nil {
		t.Fatal(err)
	}
```

Prior to bazel, the above worked okay since we run inside "cockroach" directory and we could write under pkg/ccl/importccl/testdata/...
With bazel, test runs under exec directory, with data dependencies recreated as directories with symlinks pointing back
to the actual files.  I suspect attempts to open O_CREATE|O_RDWR on a symlink is what's causing the issue.

In general, tests should *never* write anything, anywhere, except the temporary directory.

**Comment by miretskiy** [comment]
Fixed via #56127
